src/tracing/writer.rs

```diff
@@ -409,6 +415,47 @@ impl<W: Write> TraceWriter<W> {
         }
         LOG_STYLE
     }
+
+    fn write_storage_changes(&mut self, node: &CallTraceNode) -> io::Result<()> {
+        let mut changes_map = HashMap::new();
+
+        // For each call trace, compact the results so we do not write the intermediate storage
+        // writes
+        for step in &node.trace.steps {
+            if let Some(ref change) = step.storage_change {
+                let entry =
+                    changes_map.entry((&step.contract, &change.key)).or_insert((None, None));
+                if entry.0.is_none() {
+                    entry.0 = Some(change);
+                }
+                entry.1 = Some(change);
+            }
+        }
+
+        for ((contract, key), (first_change, last_change)) in changes_map {
+            if let (Some(first), Some(last)) = (first_change, last_change) {
+                let had_value = first.had_value;
+                let value = last.value;
+
+                self.write_branch()?;
+                writeln!(
+                    self.writer,
+                    " storage write {} [0x{}]:",
+                    contract.to_checksum_buffer(None).as_str(),
```

Comment:
Address is used here instead of the label. I am not sure of the best way to retrieve the label, but even if the label could be used, I personally prefer keeping the address